core-bundle/src/DependencyInjection/Compiler/AddNativeTransportFactoryPass.php

```diff
@@ -0,0 +1,44 @@
+<?php
+
+declare(strict_types=1);
+
+/*
+ * This file is part of Contao.
+ *
+ * (c) Leo Feyer
+ *
+ * @license LGPL-3.0-or-later
+ */
+
+namespace Contao\CoreBundle\DependencyInjection\Compiler;
+
+use Symfony\Component\DependencyInjection\ChildDefinition;
+use Symfony\Component\DependencyInjection\Compiler\CompilerPassInterface;
+use Symfony\Component\DependencyInjection\ContainerBuilder;
+use Symfony\Component\Mailer\Transport\NativeTransportFactory;
+
+/**
+ * @internal
+ */
+class AddNativeTransportFactoryPass implements CompilerPassInterface
+{
+    public function process(ContainerBuilder $container): void
+    {
+        if ($container->hasDefinition('mailer.transport_factory.native') || !$this->supportsNativeMailer()) {
```

Comment:
What is the advantage of encapsulating a single `class_exists()` call in a separate method? It is never re-used IIRC.